Classrooms

SteveLorentzen opened the pull request.

Commit:
made a Classrooms screen
failed to make mapping work to display list of Classroom components

trevorfehrman committed:
small typefix

Commit:
for code review

trevorfehrman commented on src/App.tsx:
```diff
@@ -6,7 +6,11 @@ import { Login } from 'components/Login'
 const App: React.FC = () => {
-  return <AuthCheck fallback={<Login />}>{<AuthenticatedApp />}</AuthCheck>
+  return (
+    <>
```
What's the point of this frag?

SteveLorentzen replied:
Good point, removed

trevorfehrman commented on src/authenticated-app.tsx:
```diff
@@ -1,52 +1,13 @@
-    console.log(values)
-  }
-
-  function validateName(value: string) {
-    console.log(value)
-    if (value === 'Steve') {
-      return 'Fuck you, Steve'
-    } else {
-      return true
-    }
-  }
```
You monster

trevorfehrman commented on src/components/Classroom.tsx:
```diff
@@ -0,0 +1,12 @@
+import * as React from 'react'
+
+type ClassroomProps = {
+  key: string
```
Unclear if needed per discussion

trevorfehrman replied:
@DoctorBurglar  did you resolve this?  we should merge this PR soon, it's got 11 commits in it

trevorfehrman commented on src/screens/Classrooms.tsx:
```diff
@@ -0,0 +1,72 @@
+  const classroomDocuments = useFirestoreCollectionData<IClassroom>(classroomRef, { idField: 'docId' })
+  console.log({ classroomDocuments, user })
+
+  async function onSubmit(values: FormData) {
```
syntax here is overkill since you're not doing anything after your await.  you may as well use the promise API and simply not invoke `then`

e.g.
`classroomRef.add({...etc}).catch(//*handle your error *//)`

SteveLorentzen replied:
That makes sense.  Will do

trevorfehrman replied:
@DoctorBurglar  done yet?

trevorfehrman commented on src/screens/Classrooms.tsx:
```diff
@@ -0,0 +1,72 @@
+          Submit
+        </Button>
+      </form>
+      {classroomDocuments.data.map((doc: IClassroom) => {
```
type not needed

SteveLorentzen replied:
Cool, fixed

trevorfehrman reviewed (changes requested):
Few minor fixes to make before approval and merge/deploy.

Commit:
working on clearing form

trevorfehrman committed:
typefix

trevorfehrman committed:
tiny improvement in code style

Commit:
fixed react-hook-form reset method usage

Commit:
Added some styling to Classroom components
Tried to change location of studentGroups to be a sub-collection of the user document on firebase
I failed :(

Commit:
updated names to use 'studentGroups' instead of 'classrooms'
attempting to store studentGroups on the user object in firebase

trevorfehrman committed:
BE model refactor to pull in student groups

trevorfehrman committed:
Merge branch 'classrooms' of https://github.com/trevorfehrman/random-name-picker into classrooms

trevorfehrman committed:
time to close this branch
